### PR TITLE
Fix perf tests (after sumMap starts to filter out -0.)

### PR DESCRIPTION
**Pull request by azat**
**Though maybe it will be better to fix sumMap for negative zeros, thoughts?**

Before perf tests was relying on the following:

    SELECT sumMap(['foo', 'bar'], [-0., -0.])

    ┌─sumMap(['foo', 'bar'], [-0., -0.])─┐
    │ (['bar','foo'],[-0,-0])            │
    └────────────────────────────────────┘

While it got changed, and now:

    ┌─sumMap(['foo', 'bar'], [-0., -0.])─┐
    │ ([],[])                            │
    └────────────────────────────────────┘

But it works for nan:

    SELECT sumMap(['foo', 'bar'], [nan, nan])

    ┌─sumMap(['foo', 'bar'], [nan, nan])─┐
    │ (['bar','foo'],[nan,nan])          │
    └────────────────────────────────────┘

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Fixes: #58959

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 44e42052b184834466970a7d95a3943eba572dec with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59287/44e42052b184834466970a7d95a3943eba572dec/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59287/44e42052b184834466970a7d95a3943eba572dec/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by azat** [comment]
>Performance Comparison [1/4] — Errors while building the report.

Likely after this change https://github.com/ClickHouse/ClickHouse/pull/57904/files#r1468474138 you cannot test changes in `compare.sh` in PRs, in other words it uses old version from upstream, so it fails again

**Review by alexey-milovidov — APPROVED** [review]

**Comment by Algunenano** [comment]
Thank you for fixing this @azat. It happened once in the PR but I couldn't reproduce it and didn't see the problem

File: tests/performance/scripts/compare.sh [hunk 1]
```diff
@@ -444,10 +444,10 @@ create view query_logs as
 create table query_run_metric_arrays engine File(TSV, 'analyze/query-run-metric-arrays.tsv')
     as
     with (
-        -- sumMapState with the list of all keys with '-0.' values. Negative zero is because
-        -- sumMap removes keys with positive zeros.
+        -- sumMapState with the list of all keys with 'nan' values. 'nan' is because
```
Comment (Algunenano) [inline tests/performance/scripts/compare.sh:447]: A simpler option is to cast the array to nullable, as `0::Nullable(Float64)` is different from the default value of Nullable(Float64) (NULL):

```
SELECT sumMap(['foo', 'foo', '4'], CAST('[-5., 5, 0]', 'Array(Nullable(Float64))')) AS t

Query id: 584265ec-d748-49c0-a4bc-4441f01f44ee

┌─t───────────────────┐
│ (['4','foo'],[0,0]) │
└─────────────────────┘

1 row in set. Elapsed: 0.001 sec.
```

Comment (azat) [reply tests/performance/scripts/compare.sh:447]: Indeed, this looks better!

Comment (Algunenano) [reply tests/performance/scripts/compare.sh:447]: Done in https://github.com/ClickHouse/ClickHouse/pull/59333

**Pull request merged**